### PR TITLE
Added Initial Code for Editing OAuth Client

### LINKED ISSUE #827: Edit/delete OAuth clients
Right now you can only create them.

### PR DESCRIPTION
**Pull request by dip-kush**
Fixes #827 

I have added the controller and edit_client page. 
I am little new to the Flask. 
I have tried to make necessary change. but the update is not happening in the Database.
@Sumukh Could you please look into my code and provide me some help. 
I will make the necessary changes and proceed with adding the delete functionality.

Thanks

### CODE REVIEW

**Comment by Sumukh** [comment]
Cool. Thanks for the PR - I'll take a look later today and leave some comments/suggestions

File: server/controllers/admin.py [hunk 1]
```diff
@@ -859,6 +859,21 @@ def clients():
...
     return render_template('staff/clients.html', clients=clients, form=form)
 
+@admin.route("/clients/<string:client_id>", methods=['GET', 'POST'])
+@is_staff()
+def client(client_id):
+    client = Client.query.get(client_id)
+    client.client_secret = utils.generate_secret_key()
+    form = forms.ClientForm(obj=client)
+    if form.validate_on_submit():
+        client = Client(user=current_user)
```
Comment (knrafto) [inline server/controllers/admin.py]: We shouldn't create a new client here, instead just use the one returned from the query above

File: server/templates/staff/clients.html [hunk 2]
```diff
@@ -34,7 +34,7 @@ <h3 class="box-title">OAuth Clients</h3>
                   {% for client in clients %}
                   <tr>
                     <td>{{ client.name }}</td>
-                    <td>{{ client.client_id }}</td>
+                    <td> <a href="{{client.client_id}}">{{ client.client_id }}</a></td>
```
Comment (knrafto) [inline server/templates/staff/clients.html]: This should be a url_for (in case we change the route)

**Comment by Sumukh** [comment]
@dip-kush - There's some comments above that should be addressed before we can merge this

**Comment by dip-kush** [comment]
I was out for a while. I will try to address the review comments soon.

**Comment by dip-kush** [comment]
@Sumukh @knrafto  I have made changes mentioned in the review comments.

**Pull request merged**